mne/viz/ica.py

```diff
@@ -130,6 +132,257 @@ def plot_ica_sources(ica, inst, picks=None, exclude=None, start=None,
+    # the rest is component-specific
+    for idx, pick in enumerate(picks):
+        if idx > 0:
+            fig, axes = _create_properties_layout(figsize=figsize)
+
+        # spectrum
+        this_psd = psds[:, idx, :]
+        if dB:
+            this_psd = 10 * np.log10(this_psd)
+        psds_mean = this_psd.mean(axis=0)
+        diffs = this_psd - psds_mean
+        # the distribution of power for each frequency bin is highly
+        # skewed so we calculate std for values below and above average
+        # separately - this is used for fill_between shade
+        spectrum_std = [
+            [np.sqrt((d[d < 0] ** 2).mean(axis=0)) for d in diffs.T],
+            [np.sqrt((d[d > 0] ** 2).mean(axis=0)) for d in diffs.T]]
+        spectrum_std = np.array(spectrum_std) * num_std
+
+        # erp std
+        if plot_std:
+            erp = ica_data[idx].mean(axis=0)
+            diffs = ica_data[idx] - erp
+            erp_std = [
+                [np.sqrt((d[d < 0] ** 2).mean(axis=0)) for d in diffs.T],
+                [np.sqrt((d[d > 0] ** 2).mean(axis=0)) for d in diffs.T]]
+            erp_std = np.array(erp_std) * num_std
+
+        # epoch variance
+        epoch_var = np.var(ica_data[idx], axis=1)
+
+        # plotting
+        # --------
+        # component topomap
+        _plot_ica_topomap(ica, pick, show=False, axis=axes[0], **topomap_args)
+
+        # image and erp
+        plot_epochs_image(epochs_src, picks=pick, axes=axes[1:3],
+                          colorbar=False, show=False, **image_args)
+
+        # spectrum
+        axes[3].plot(freqs, psds_mean, color='k')
+        if plot_std:
+            axes[3].fill_between(freqs, psds_mean - spectrum_std[0],
+                                 psds_mean + spectrum_std[1],
+                                 color='k', alpha=.15)
+        if plot_lowpass_edge:
+            axes[3].axvline(inst.info['lowpass'], lw=2, linestyle='--',
+                            color='k', alpha=0.15)
+
+        # epoch variance
+        axes[4].scatter(range(len(epoch_var)), epoch_var, alpha=0.5,
+                        facecolor=[0, 0, 0], lw=0)
+
+        # aesthetics
+        # ----------
+        axes[0].set_title('IC #{:0>3}'.format(str(pick)))
+
+        set_title_and_labels(axes[1], 'epochs image and ERP/ERF', [], 'Epochs')
+
+        # erp
+        set_title_and_labels(axes[2], [], 'time', 'AU')
+        # line color and std
+        axes[2].lines[0].set_color('k')
+        if plot_std:
+            erp_xdata = axes[2].lines[0].get_data()[0]
+            axes[2].fill_between(erp_xdata, erp - erp_std[0],
+                                 erp + erp_std[1], color='k', alpha=.15)
+            axes[2].autoscale(enable=True, axis='y')
+            axes[2].axis('auto')
+            axes[2].set_xlim(erp_xdata[[0, -1]])
+        # remove half of yticks if more than 5
+        yt = axes[2].get_yticks()
+        if len(yt) > 5:
+            yt = yt[::2]
+            axes[2].set_yticks(yt)
```

Comment:
I think this will raise a deprecation warning on mpl. -> ``axes[2].yaxis.set_ticks(yt)``.
I changed it in ``_hide_frame`` in my pending PR so don't worry about changing it there.